Tests/UnitTests/ess-upload-file.component.spec.ts

```diff
@@ -74,69 +91,103 @@ const getNoEncData = () => {
+    `('processEncFile should set raise "Some values have not been added to list." info',
+    ({ fileType, fileName, getEncData, encDataFunc, expectedResult }: { fileType: 'text/csv' | 'text/permit'; fileName: string; getEncData: string; encDataFunc: string; expectedResult: number }) => {
+      const file = new File([getEncData], fileName);
+      Object.defineProperty(file, 'type', { value: fileType });
       component.encFile = file;
-      component.processEncFile(getInvalidEncData());
-      expect(component.validEncList.length).toEqual(1);
+      component.processEncFile(encDataFunc);
+      expect(component.validEncList.length).toEqual(expectedResult);
       expect(component.messageType).toEqual('info');
       expect(component.messageDesc).toEqual('Some values have not been added to list.');
     });
-    it('processEncFile should set raise "No ENCs found." info',() => {
-      const file = new File([getEncData()], 'test.txt');
-      Object.defineProperty(file, 'type', { value: 'text/plain' });
+
+  it.each`
+    fileType           |fileName         | getEncData              | encDataFunc                 | expectedResult
+    ${'text/csv'}      |${'test.csv'}    | ${getEncData_csv()}     | ${getNoEncData_csv()}  |  ${3}
+    ${'text/plain'}    |${'test.txt'}    | ${getEncData()}         | ${getNoEncData()}      |  ${1}
+    `('processEncFile should set raise "No ENCs found." info',
+    ({ fileType, fileName, getEncData, encDataFunc, expectedResult }: { fileType: 'text/csv' | 'text/permit'; fileName: string; getEncData: string; encDataFunc: string; expectedResult: number }) => {
+      const file = new File([getEncData], fileName);
+      Object.defineProperty(file, 'type', { value: fileType });
       component.encFile = file;
-      component.processEncFile(getNoEncData());
-      expect(component.validEncList.length).toEqual(0);
+      component.processEncFile(encDataFunc);
+      expect(component.validEncList).toEqual([]);
       expect(component.messageType).toEqual('info');
       expect(component.messageDesc).toEqual('No ENCs found.');
     });
 
-    it('uploadListener should raise error if file is not provided' , () =>{
-      const file = new File([getEncData()], 'test.jpeg');
-      Object.defineProperty(file, 'size', { value: 1024 * 1024 + 1 });
-      Object.defineProperty(file, 'type', { value: 'image/jpeg' });
-      const event = {
-        srcElement : {
-          files : [file]
-        }
-      };
-      expect(component.displayErrorMessage).toBe(false);
-      component.uploadListener(event);
-      expect(component.validEncList.length).toEqual(0);
-      expect(component.messageType).toEqual('error');
-      expect(component.messageDesc).toEqual('Please select a .csv or .txt file');
-      expect(component.displayErrorMessage).toBe(true);
-    });
+  it('uploadListener should raise error if file is not provided', () => {
```

Comment:
I think the name of this test is misleading. Please change to `uploadListener should raise error for unsupported file type`
```suggestion
  it('uploadListener should raise error for unsupported file type', () => {
```

Reply:
Comment updated. Thanks.